assets/src/edit-story/app/history/test/reducer.js

```diff
@@ -0,0 +1,250 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+  default as useReducer,
+  ADD_ENTRY,
+  CLEAR_HISTORY,
+  CLEAR_REPLAY_STATE,
+  EMPTY_STATE,
+  REPLAY,
+} from '../reducer';
+
+describe('reducer', () => {
+  const size = 5;
+  const reducer = useReducer(size);
+  it('should throw error if unknown action given', () => {
+    expect(() =>
+      reducer(EMPTY_STATE, {
+        type: 'UNKNOWN_ACTION',
+        payload: {},
+      })
+    ).toThrow(/Unknown history reducer action: UNKNOWN_ACTION/i);
+  });
+
+  describe('addEntry', () => {
+    it('should add entry to the beginning of the history', () => {
+      const initialState = {
+        ...EMPTY_STATE,
+        entries: [{ id: 1 }],
+      };
+
+      const newEntry = { id: 2 };
+      const result = reducer(initialState, {
+        type: ADD_ENTRY,
+        payload: newEntry,
+      });
+
+      expect(result.entries[0]).toStrictEqual(newEntry);
+    });
+
+    it('should clear undone states when adding a new entry', () => {
+      const initialState = {
+        ...EMPTY_STATE,
+        entries: [{ id: 1 }, { id: 2 }, { id: 3 }],
+        offset: 2,
+      };
+      const newEntry = { id: 4 };
+      const result = reducer(initialState, {
+        type: ADD_ENTRY,
+        payload: newEntry,
+      });
+
+      expect(result.entries).toHaveLength(2);
+      expect(result.entries[1].id).toStrictEqual(3);
+      expect(result.entries[0].id).toStrictEqual(4);
+    });
+
+    it('should set correct state values when adding new entry', () => {
+      const initialState = {
+        ...EMPTY_STATE,
+        versionNumber: 3,
+        entries: [{ id: 1 }],
+        offset: 2,
+      };
+      const newEntry = { id: 2 };
+      const result = reducer(initialState, {
+        type: ADD_ENTRY,
+        payload: newEntry,
+      });
+
+      expect(result.versionNumber).toStrictEqual(4);
+      expect(result.offset).toStrictEqual(0);
+    });
+
+    it('should respect the determined history size when adding a new entry', () => {
+      const entries = [...Array(size + 1)].map(() => {
+        return { id: 'foo' };
+      });
+      const initialState = {
+        ...EMPTY_STATE,
+        entries,
+      };
+      const newEntry = { id: 7 };
+      const result = reducer(initialState, {
+        type: ADD_ENTRY,
+        payload: newEntry,
+      });
+      expect(result.entries).toHaveLength(size);
+    });
+  });
+
+  describe('clearHistory', () => {
+    it('should set initial state when clearing history', () => {
+      const initialState = {
+        offset: 1,
+        replayState: { id: 1 },
+        entries: [{ id: 1 }],
+        versionNumber: 1,
+      };
+
+      const result = reducer(initialState, {
+        type: CLEAR_HISTORY,
+      });
+
+      Object.keys(result).forEach((k) => {
+        expect(result[k]).toStrictEqual(EMPTY_STATE[k]);
+      });
```

Comment:
Couldn't this be done with a regular `.toMatchObject`?

Reply:
Yes, it can! Changed.